ansible_ai_connect/ai/api/serializers.py

```diff
@@ -472,12 +472,87 @@ def validate(self, data):
         return data
 
 
+class GenerationRoleRequestSerializer(serializers.Serializer):
+    class Meta:
+        fields = [
+            "text",
+            "outline",
+            "createOutline",
+            "additionalContext",
+            "fileTypes",
+            "generationId",
+            "wizardId",
+        ]
+
+    text = AnonymizedCharField(
+        required=True,
+        label="the goal of the role",
+        help_text=(
+            "Can be in one or a few sentences. WCA will do HAP filtering "
+            "as appropriate, and reject input above a certain HAP threshold."
+        ),
+    )
+    outline = AnonymizedCharField(
+        required=False,
+        label="an outline of the role",
+        help_text="An outline of the role should be a numbered list.",
+    )
+    createOutline = serializers.BooleanField(
+        required=False,
+        default=False,
+        label="generate outline",
+        help_text=(
+            "Indicates whether the answer should also include an outline " "of the Ansible Role."
+        ),
+    )
+    additionalContext = serializers.DictField(
+        required=False,
+        allow_empty=True,
+        label="inline suggestions",
+        help_text=("Parameter use for the inline suggestions."),
+    )
+    fileTypes = serializers.ListField(
+        required=False,
+        child=serializers.CharField(),
+        allow_empty=False,
+        label="file types",
+        help_text=(
+            "The file types generated by the model. Default is ['task', 'default']. "
+            "The file type name is based on the inner role directories, "
+            "without the trailing 's'"
+        ),
+    )
+    generationId = serializers.UUIDField(
+        format="hex_verbose",
+        required=False,
+        label="generation ID",
+        help_text=("A UUID that identifies the particular generation data is being requested for."),
+    )
+    wizardId = serializers.UUIDField(
+        format="hex_verbose",
+        required=False,
+        label="wizard ID",
+        help_text=("A UUID to track the succession of interaction from the user."),
+    )
+
+    metadata = Metadata(required=False)
+
+    def validate(self, data):
+        return data
+
+
 class GenerationWarningResponseSerializer(serializers.Serializer):
     id = serializers.CharField()
     message = serializers.CharField()
     details = serializers.CharField(required=False)
 
 
+class GenerationRoleFileEntrySerializer(serializers.Serializer):
+    path = serializers.CharField()
+    content = serializers.CharField()
+    file_type = serializers.CharField()
```

Comment:
It should be called `fileType` if we want to continue to respect the camelCase convention.